torch_em/data/datasets/electron_microscopy/cellmap.py

```diff
@@ -0,0 +1,339 @@
+        test_crops = get_test_crops()
+        log.info(f"Found '{len(test_crops)}' test crops.")
+
+    # Fetch all the crop manifests.
+    if crops == "all":
+        crops_parsed = crops_from_manifest + test_crops
+    elif crops == "test":
+        crops_parsed = test_crops
+    else:  # Otherwise, custom crops are parsed.
+        crops_split = tuple(int(x) for x in crops.split(","))
+        crops_parsed = tuple(filter(lambda v: v.id in crops_split, crops_from_manifest))
+
+    # Now get the crop ids.
+    if len(crops_parsed) == 0:
+        log.info(f"No crops found matching '{crops}'. Doing nothing.")
+        return
+
+    crop_ids = tuple(c.id for c in crops_parsed)
+    log.info(f"Preparing to copy the following crops: '{crop_ids}'.")
+    log.info(f"Data will be saved to '{dest_path_abs}'.")
+
+    for crop in crops_parsed:
+        log = log.bind(crop_id=crop.id, dataset=crop.dataset)
+
+        # Check whether the crop path has been downloaded already or not.
+        crop_path = dest_path_abs / f"crop_{crop.id}.h5"
+        if crop_path.exists():
+            log.info(f"The crop '{crop.id}' is already saved at '{crop_path}'.")
+            log = log.unbind("crop_id", "dataset")
+            continue
+
+        # If 'download' is set to 'False', we do not go further from here.
+        if not download:
+            log.error(f"Cannot download the crop '{crop.id}' as 'download' is set to 'False'.")
+            return
+
+        # Get the EM volume of highest resolution.
+        em_source_group = read_group(str(crop.em_url), storage_options={"anon": True})
+        log.info(f"Found EM data at {crop.em_url}.")
+
+        # Let's get the multiscale model of the source em group.
+        array_wrapper = {"name": "dask_array", "config": {"chunks": "auto"}}
+        em_source_arrays = read_multiscale_group(em_source_group, array_wrapper)
+        em_s0_array = em_source_arrays[resolution]
+
+        # Get the ground-truth (gt) masks.
+        gt_source_group = read_group(str(crop.gt_source), storage_options={"anon": True})
+
+        # Let's get all ground-truth hierarchies.
+        # NOTE: Following the same as the original repo, relying on fs.find to avoid slowness in traversing online zarr.
+        fs = gt_source_group.store.fs
+        store_path = gt_source_group.store.path
+        gt_files = fs.find(store_path)
+
+        crop_group_inventory = tuple(fn.split(store_path)[-1] for fn in gt_files)
+        crop_group_inventory = tuple(curr_cg[1:].split("/")[0] for curr_cg in crop_group_inventory)
+        crop_group_inventory = np.unique(crop_group_inventory).tolist()
+        crop_group_inventory = [curr_cg for curr_cg in crop_group_inventory if curr_cg not in [".zattrs", ".zgroup"]]
+
+        gt_crop_shape = gt_source_group[f"all/{resolution}"].shape  # since "all" exists "al"ways, we rely on it.
+
+        # Get the offset values for the ground truth crops.
+        for _, group in gt_source_group.groups():
```

Comment:
Do we always get the lowest resolution group with this? It looks a bit dangerous to rely on this ...

Reply:
Ok, nvm, it looks like this is happening below (line 139). But the code here is still a bit unclear to read. Maybe another comment on what's going on would help.